### PR TITLE
Some 1.2 relnotes

### PR DESCRIPTION
**Pull request by brson**
Haven't finished these, but since [others](https://github.com/rust-lang/rust/pull/26579) are also working on this and I'm on vacation I figured I should get this in.

### CODE REVIEW

File: AUTHORS.txt [hunk 1]
```diff
@@ -760,10 +813,13 @@ nsf <no.smile.face@gmail.com>
 nwin <nwin@users.noreply.github.com>
 Oak <White-Oak@users.noreply.github.com>
 OGINO Masanori <masanori.ogino@gmail.com>
+OlegTsyba <idethrone1@gmail.com>
+Oliver Schneider <git1984941651981@oli-obk.de>
```
Comment (alexcrichton) [inline AUTHORS.txt]: Candidate for a mailmap entry?

File: AUTHORS.txt [hunk 2]
```diff
@@ -786,7 +843,9 @@ Paul Banks <banks@banksdesigns.co.uk>
 Paul Collier <paul@paulcollier.ca>
 Paul Collins <paul@ondioline.org>
 Paul Crowley <paulcrowley@google.com>
+Paul Faria <Nashenas88@gmail.com>
```
Comment (alexcrichton) [inline AUTHORS.txt]: Another candidate for a mailmap entry perhaps?

File: AUTHORS.txt [hunk 3]
```diff
@@ -795,9 +854,11 @@ Pavel Panchekha <me@pavpanchekha.com>
 Pawel Olzacki <p.olzacki2@samsung.com>
 Pedro Larroy <pedro.larroy@here.com>
 Peer Aramillo Irizar <peer.aramillo.irizar@gmail.com>
+peferron <pe.ferron@gmail.com>
 Pete Hunt <petehunt@users.noreply.github.com>
 Peter Atashian <retep998@gmail.com>
 Peter Elmers <peter.elmers@rice.edu>
+Peter Elmers <peter.elmers@yahoo.com>
```
Comment (alexcrichton) [inline AUTHORS.txt]: (mailmap)

File: AUTHORS.txt [hunk 4]
```diff
@@ -1058,12 +1134,15 @@ Will <will@glozer.net>
 Wojciech Ogrodowczyk <github@haikuco.de>
 wonyong kim <wonyong.kim@samsung.com>
 xales <xales@naveria.com>
+Xuefeng Wu <benewu@gmail.com>
+XuefengWu <benewu@gmail.com>
```
Comment (alexcrichton) [inline AUTHORS.txt]: (mailmap above and below I think)

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,110 @@
...
+  methods that convert case, following Unicode case mapping.
+* It is now easier to handle to poisoned locks. The [`PoisonError`]
+  type, returned by failing lock operations, exposes `into_inner`,
+  `get_ref`, and `get_mut`, which all give access to the inner lock
+  guard, and allow the poisoned lock to continue to operate. The
+  `is_poisoned` method of [`RwLock`] and [`Mutex`] can poll for a
+  poisoned lock without attempting to take the lock.
+* On Unix the [`FromRawFd`] trait is implemented for [`Stdio`], and
+  [`AsRawFd`] for [`ChildStdin`], [`ChildStdout`], [`ChildStderr`].
+  On Windows the FromRawHandle` trait is implemented for `Stdio`,
```
Comment (alexcrichton) [inline RELEASES.md]: I think the backticks here are off-by-one

**Comment by alexcrichton** [comment]
Nice! r=me with a few nits

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,110 @@
...
+  [`wrapping_neg`], [`wrapping_shl`], [`wrapping_shr`]. These are in
+  addition to the existing [`wrapping_add`], [`wrapping_sub`], and
+  [`wrapping_mul`] methods, and alternatives to the [`Wrapping`]
+  type.. It is illegal for the default arithmetic operations in Rust
+  to overflow; the desire to wrap must be explicit.
+* [`fmt::Formatter`] implements [`fmt::Write`], a `fmt`-specific trait
+  for writing data to formatted strings, similar to [`io::Write`].
+* [`fmt::Formatter`] adds 'debug builder' methods, [`debug_struct`],
+  [`debug_tuple`], [`debug_list`], [`debug_set`], [`debug_map`]. These
+  are used by code generators to emit implementations of [`Debug`].
```
Comment (bluss) [inline RELEASES.md]: Maybe this is the right place to mention `{:#?}` (pretty printing by using alternative format for Debug). It's hugely useful so we need to document it. It's a debug printer feature, only problem it was available already from Rust 1.1 in derived Debug impls.

**Comment by bluss** [comment]
I tagged PR #26058, so its entry could read
```
[Language]
* Patterns with `ref mut` now correctly invoke [`DerefMut`] when matching against dereferencable values.

[`DerefMut`]: http://doc.rust-lang.org/nightly/std/ops/trait.DerefMut.html
```

File: .mailmap [hunk 7]
```diff
@@ -13,7 +13,7 @@ Alex Rønne Petersen <alex@lycus.org>
 Andreas Gal <gal@mozilla.com> <andreas.gal@gmail.com>
 Andrew Poelstra <asp11@sfu.ca> <apoelstra@wpsoftware.net>
 Anton Löfgren <anton.lofgren@gmail.com> <alofgren@op5.com>
-Ariel Ben-Yehuda <arielb1@mail.tau.ac.il> <ariel.byd@gmail.com>
+Ariel Ben-Yehuda <arielb1@mail.tau.ac.il> arielb1 <arielb1@mail.tau.ac.il> <ariel.byd@gmail.com>
```
Comment (arielb1) [inline .mailmap]: There's no need to have my mail twice.

**Pull request merged**